packages/rs-drive-abci/src/execution/engine/run_block_proposal/v1/mod.rs

```diff
@@ -0,0 +1,427 @@
+use dpp::block::epoch::Epoch;
+
+use dpp::validation::ValidationResult;
+use drive::error::Error::GroveDB;
+
+use dpp::version::PlatformVersion;
+use drive::grovedb::Transaction;
+
+use crate::abci::AbciError;
+use crate::error::execution::ExecutionError;
+
+use crate::error::Error;
+use crate::execution::types::block_execution_context::v0::{
+    BlockExecutionContextV0Getters, BlockExecutionContextV0MutableGetters,
+};
+use crate::execution::types::block_execution_context::BlockExecutionContext;
+use crate::execution::types::block_fees::v0::BlockFeesV0;
+use crate::execution::types::block_state_info::v0::{
+    BlockStateInfoV0Getters, BlockStateInfoV0Methods, BlockStateInfoV0Setters,
+};
+use crate::execution::types::{block_execution_context, block_state_info};
+use crate::metrics::HistogramTiming;
+use crate::platform_types::block_execution_outcome;
+use crate::platform_types::block_proposal;
+use crate::platform_types::epoch_info::v0::{EpochInfoV0Getters, EpochInfoV0Methods};
+use crate::platform_types::epoch_info::EpochInfo;
+use crate::platform_types::platform::Platform;
+use crate::platform_types::platform_state::v0::PlatformStateV0Methods;
+use crate::platform_types::platform_state::PlatformState;
+use crate::platform_types::verify_chain_lock_result::v0::VerifyChainLockResult;
+use crate::rpc::core::CoreRPCLike;
+
+impl<C> Platform<C>
+where
+    C: CoreRPCLike,
+{
+    /// Runs a block proposal, either from process proposal or prepare proposal.
+    ///
+    /// This function takes a `BlockProposal` and a `Transaction` as input and processes the block
+    /// proposal. It first validates the block proposal and then processes raw state transitions,
+    /// withdrawal transactions, and block fees. It also updates the validator set.
+    ///
+    /// # Arguments
+    ///
+    /// * `block_proposal` - The block proposal to be processed.
+    /// * `known_from_us` - Do we know that we made this block proposal?.
+    /// * `transaction` - The transaction associated with the block proposal.
+    ///
+    /// # Returns
+    ///
+    /// * `Result<ValidationResult<BlockExecutionOutcome, Error>, Error>` - If the block proposal is
+    ///   successfully processed, it returns a `ValidationResult` containing the `BlockExecutionOutcome`.
+    ///   If the block proposal processing fails, it returns an `Error`. Consensus errors are returned
+    ///   in the `ValidationResult`, while critical system errors are returned in the `Result`.
+    ///
+    /// # Errors
+    ///
+    /// This function may return an `Error` variant if there is a problem with processing the block
+    /// proposal, updating the core info, processing raw state transitions, or processing block fees.
+    ///
+    pub(super) fn run_block_proposal_v1(
```

Comment:
Call v0 instead of copy-paste?

Reply:
This is not possible. As `run_block_proposal_v1` performs an extra operation in the middle of `run_block_proposal_v0` logic.